README.md

```diff
@@ -180,7 +180,7 @@ The following shows the techniqual details:
 |---|---|---|---|
 |wasmtime|:heavy_check_mark:|:heavy_check_mark:||
 |wamr(wasm-micro-runtime)|:heavy_check_mark:|:heavy_check_mark:||
-|wazero|:construction: (stdin unsupported)|:heavy_check_mark:|non-blocking stdin doesn't seem to work|
+|wazero|:heavy_check_mark:|:heavy_check_mark:||
```

Comment:
ps this is hoping things work, if CI or manual verification say otherwise, I'll happily revert it! well.. I will not be gleeful, but will politely revert it ;)

Reply:
@codefromthecrypt  Thank you for your work on wazero and patches for our project! 
IIUC, as of now, wazero doesn't seem to support poll_oneoff (i.e. `select()` on wasi-sdk) for stdin? (Though it now returns the EBADF which is compatible to wasi-sdk) https://github.com/tetratelabs/wazero/blob/v1.0.1/imports/wasi_snapshot_preview1/poll.go#L135-L137 Our stdin implementation relies on poll_oneoff with support at least for stdin fd.

Reply:
nm I see the build failed. I think we must have fixed something similar, but not this. @evacchi can you look at the failure here and make sure we have a test in wazero that fails the same way (or an issue with it)? I will revert the code and README change as we still aren't quite there, yet.

Reply:
@codefromthecrypt Thanks for the reply. I meant that `select` dones't detect inputs for stdin fd on wazero. I'm not familiar to wazero so maybe I'm missing something on the configuration? 

This is the reproducer:

sample c program:

```c
#include <stdio.h>
#include <unistd.h>
#include <sys/select.h>

int main(int argc, char **argv)
{
  char buf[5];
  int len = 5, ret;
  fd_set rfds;
  struct timeval tv;

  FD_ZERO(&rfds);
  FD_SET(0, &rfds);

  tv.tv_sec = 3;
  tv.tv_usec = 0;
  printf("waiting for 3s for inputs...\n");
  ret = select(1, &rfds, NULL, NULL, &tv);
  if ((ret > 0) && FD_ISSET(0, &rfds)) {
    ret = read(0, buf, len);
    if (ret > 0) {
      printf("got %.*s (first 5 chars)\n", ret, buf);
    }
  } else {
    printf("no input\n");
  }
}
```

wazero code:

```go
package main

import (
	"context"
	"os"

	"github.com/tetratelabs/wazero"
	"github.com/tetratelabs/wazero/imports/wasi_snapshot_preview1"
)

func main() {
	ctx := context.Background()
	c, err := os.ReadFile(os.Args[1])
	if err != nil {
		panic(err)
	}
	r := wazero.NewRuntime(ctx)
	defer func() {
		r.Close(ctx)
	}()
	wasi_snapshot_preview1.MustInstantiate(ctx, r)
	compiled, err := r.CompileModule(ctx, c)
	if err != nil {
		panic(err)
	}
	_, err = r.InstantiateModule(ctx, compiled, wazero.NewModuleConfig().WithStdout(os.Stdout).WithStderr(os.Stderr).WithStdin(os.Stdin))
	if err != nil {
		panic(err)
	}
	return
}
```

wasm file is built using wasi-sdk 19.0: [out.tar.gz](https://github.com/ktock/container2wasm/files/11109627/out.tar.gz)

```
clang --sysroot=${WASI_SDK_PATH}/share/wasi-sysroot -o /out.wasm /main.c
```

On host, it works properly. It waits for inputs for 3s and prints the input string.

```console
$ gcc main.c && ./a.out
waiting for inputs...
hi
got hi
 (first 5 chars)
```

On wazero, `select()` doesn't wait nor detect for input and returns immediately.

```console
$ go run main.go ./out/out.wasm
waiting for inputs...
no input
```

Reply:
I am on it!

Reply:
thanks!